Src/LinearSolvers/MLMG/AMReX_MLEBTensor_2D_K.H

```diff
@@ -40,12 +42,24 @@ void mlebtensor_cross_terms_fx (Box const& box, Array4<Real> const& fx,
                 int jhim = j - flag(i  ,j,0).isConnected(0,-1,0);
                 int jlop = j + flag(i-1,j,0).isConnected(0, 1,0);
                 int jlom = j - flag(i-1,j,0).isConnected(0,-1,0);
+
                 Real whi = mlebtensor_weight(jhip-jhim);
                 Real wlo = mlebtensor_weight(jlop-jlom);
-                Real dudy = (0.5*dyi) * ((vel(i  ,jhip,0,0)-vel(i  ,jhim,0,0))*whi
-                                        +(vel(i-1,jlop,0,0)-vel(i-1,jlom,0,0))*wlo);
-                Real dvdy = (0.5*dyi) * ((vel(i  ,jhip,0,1)-vel(i  ,jhim,0,1))*whi
-                                        +(vel(i-1,jlop,0,1)-vel(i-1,jlom,0,1))*wlo);
+
+                Real aux = 1.0;
+                if (i==domain.hiVect()[0]+1 and bct[2*AMREX_SPACEDIM+0]==AMREX_LO_NEUMANN){
+                    whi = 0.0;
+                    aux = 2.0;
+                }
+                if (i==domain.loVect()[0] and bct[0*AMREX_SPACEDIM+1]==AMREX_LO_NEUMANN){
+                    wlo = 0.0;
+                    aux = 2.0;
+                }
+
+                Real dudy = (0.5*dyi*aux) * ((vel(i  ,jhip,0,0)-vel(i  ,jhim,0,0))*whi
+                                            +(vel(i-1,jlop,0,0)-vel(i-1,jlom,0,0))*wlo);
+                Real dvdy = (0.5*dyi*aux) * ((vel(i  ,jhip,0,1)-vel(i  ,jhim,0,1))*whi
+                                            +(vel(i-1,jlop,0,1)-vel(i-1,jlom,0,1))*wlo);
```

Comment:
I am not sure this is correct.  If I understand this, `bct` is an array that is supposed to be used like `bct[face*AMREX_SPACEDIM+icomp]`.  For `dudy`, `icomp = 0` and for `dvdy`, `icomp = 1`.  But in the code above, they use the same `aux` obtained from the same `icomp`.  Also we should use `Orientation` instead of `int` so that it's easy to read.  We might also want to use `Array2D<0,AMREX_SPACEDIM-1,0,AMREX_SPACEDIM-1>`.  Then we can write `bct[Orientation(Direction::x,Orientation::low)][0]`.  Note we still need to make sure the order of the 2d array here is consistent with where it is set before kernel launch.

Reply:
Thanks for the review!! I will fix this!!